code/modules/research/designs.dm

```diff
@@ -548,27 +533,11 @@
 	build_path = /obj/item/device/flash/advanced
 	sort_string = "TADAA"
 
-/datum/design/item/weapon/stunrevolver
-	name = "stun revolver"
-	id = "stunrevolver"
-	req_tech = list(TECH_COMBAT = 3, TECH_MATERIAL = 3, TECH_POWER = 2)
-	materials = list(MATERIAL_STEEL = 4000)
-	build_path = /obj/item/weapon/gun/energy/stunrevolver
-	sort_string = "TADAB"
-
-/datum/design/item/weapon/stunrifle
-	name = "stun rifle"
-	id = "stun_rifle"
-	req_tech = list(TECH_COMBAT = 4, TECH_MATERIAL = 3, TECH_POWER = 3)
-	materials = list(MATERIAL_STEEL = 4000, MATERIAL_GLASS = 1000, MATERIAL_SILVER = 500)
-	build_path = /obj/item/weapon/gun/energy/stunrevolver/rifle
-	sort_string = "TADAC"
-
 /datum/design/item/weapon/contactbeam
 	name = "C99 Supercollider Contact Beam"
 	id = "contactbeam"
 	req_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 5, TECH_POWER = 3)
-	materials = list(MATERIAL_STEEL = 5000, MATERIAL_GLASS = 1000, "diamond" = 400)
+	materials = list(MATERIAL_STEEL = 2500, MATERIAL_GOLD = 1500, "uranium" = 800)
```

Comment:
This shouldn't require different materials to the battery. Needing a total of four rare materials to get a working contact beam is hella excessive. make the gun and battery use the same ingredients in different quantities

gold and uranium seem like the best choice honestly, radioactivity and conductivity fit an energy weapon, so i'd say change the battery requirements to use them instead of plasteel and diamond

Reply:
Oops... Must've screwed something up here. I never intended them to use two different ones. I'll mark the required items for them as gold and uranium, with the ammo costing marginally less.